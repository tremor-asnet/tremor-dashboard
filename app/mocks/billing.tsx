// Constants
import { AGGREGATION_TYPE } from "@/constants";

export const mockSalaryData = [
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
];

export const mockBillingCard = {
  id: 1,
  type: "Mastercard",
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};
