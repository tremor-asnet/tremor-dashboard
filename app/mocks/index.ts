export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
