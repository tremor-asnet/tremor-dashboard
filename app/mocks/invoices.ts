import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    date: "March, 21, 2021",
    code: "MS-123456",
    price: 180,
  },
  {
    date: "Apr, 22, 2022",
    code: "MS-123457",
    price: 180,
  },
  {
    date: "May, 23, 2023",
    code: "MS-999999",
    price: 180,
  },
  {
    date: "Jun, 24, 2024",
    code: "MS-888888",
    price: 180,
  },
  {
    date: "Jul, 25, 2025",
    code: "MS-476543",
    price: 180,
  },
];
