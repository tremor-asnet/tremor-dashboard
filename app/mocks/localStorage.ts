// Mock localStorage
export const mockLocalStorage = (() => {
  let store: Record<string, string> = {};

  return {
    getItem: jest.fn((key: string) => store[key] || null),
    setItem: jest.fn((key: string, value: string) => {
      store[key] = value.toString();
    }),
    removeItem: jest.fn((key: string) => {
      delete store[key];
    }),
    clear: jest.fn(() => {
      store = {};
    }),
  };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });
