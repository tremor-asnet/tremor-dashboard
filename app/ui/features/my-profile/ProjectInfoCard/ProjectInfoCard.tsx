// Components
import Link from "next/link";
import Image from "next/image";
import { Title, Text, Flex } from "@tremor/react";
import { Avatar, Button } from "@/ui/components";

// Constants
import { ROUTES } from "@/constants/routes";
import { VARIANT_BUTTON } from "@/constants";

interface AvatarProps {
  name: string;
  avatar: string;
}

interface ProjectCard {
  id: number;
  cover: string;
  primaryName: string;
  description: string;
  secondaryName: string;
  participants: AvatarProps[];
}

interface ProjectInfoCardProps {
  links: ProjectCard[];
}

export const ProjectInfoCard = ({ links }: ProjectInfoCardProps) => {
  const renderProjectCards = links.map(
    ({ id, cover, description, participants, primaryName, secondaryName }) => (
      <Flex
        key={id}
        justifyContent="start"
        flexDirection="col"
        alignItems="start">
        <Flex>
          <Image
            className="w-full rounded-xl shadow-lg"
            src={cover}
            width={315}
            height={180}
            alt={primaryName}
          />
        </Flex>
        <Flex
          justifyContent="start"
          flexDirection="col"
          alignItems="start"
          className="pt-4 pl-2">
          <Text className="capitalize font-normal dark:text-dark-tremor-content-romance">
            {secondaryName}
          </Text>
          <Title className="w-full font-bold text-xl capitalize mb-2 text-primary dark:text-dark-primary">
            <Link href={ROUTES.PROJECTS}>{primaryName}</Link>
          </Title>
          <Text className="flex-wrap font-light mb-6 dark:text-dark-tremor-content-romance">
            {description}
          </Text>
          <Flex>
            <Link href={ROUTES.PROJECTS}>
              <Button
                additionalClass="font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
                variant={VARIANT_BUTTON.SECONDARY_SHADOW}
                variantTremor={VARIANT_BUTTON.SECONDARY}>
                <Text className="font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom">
                  View Project
                </Text>
              </Button>
            </Link>
            <div className="container-avatars min-w-[56px] flex justify-end flex-nowrap ml-3">
              {participants.map(({ name, avatar }) => (
                <Avatar
                  key={`${name}-${id}`}
                  alt={name}
                  className="relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
                  width={20}
                  height={20}
                  priority
                  src={avatar}
                  sizes="100%"
                />
              ))}
            </div>
          </Flex>
        </Flex>
      </Flex>
    ),
  );

  return (
    <Flex
      justifyContent="start"
      alignItems="start"
      className="flex-wrap gap-y-12 md:grid md:grid-cols-2 md:gap-8 xl:grid-cols-3 xl:gap-8 2xl:grid-cols-4">
      {renderProjectCards}
    </Flex>
  );
};

export default ProjectInfoCard;
