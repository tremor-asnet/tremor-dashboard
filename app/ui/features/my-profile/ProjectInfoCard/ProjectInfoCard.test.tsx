// ProjectInfoCard.test.tsx
import { render } from "@testing-library/react";

// Components
import ProjectInfoCard from "./ProjectInfoCard";

// Mocks
import { PROFILE_INFO_PROJECT_CARD } from "@/mocks";

const ProjectInfoCardProps = {
  links: PROFILE_INFO_PROJECT_CARD,
};

const ProjectInfoCardComponent = () =>
  render(<ProjectInfoCard {...ProjectInfoCardProps} />);

describe("ProjectInfoCard Component", () => {
  it("render ProjectInfoCard component with snapshot correctly", () => {
    const { container } = ProjectInfoCardComponent();

    expect(container).toMatchSnapshot();
  });

  it("renders ProjectInfoCard component with mock data", () => {
    const { getByText } = ProjectInfoCardComponent();

    const projectNames = PROFILE_INFO_PROJECT_CARD.map(({ primaryName }) =>
      getByText(primaryName),
    );
    expect(projectNames.length).toBe(PROFILE_INFO_PROJECT_CARD.length);
  });
});
