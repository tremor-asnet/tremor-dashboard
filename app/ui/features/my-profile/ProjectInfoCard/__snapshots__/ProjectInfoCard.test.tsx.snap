// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInfoCard Component render ProjectInfoCard component with snapshot correctly 1`] = `
<div>
  <div
    class="tremor-Flex-root flex w-full flex-row justify-start items-start flex-wrap gap-y-12 md:grid md:grid-cols-2 md:gap-8 xl:grid-cols-3 xl:gap-8 2xl:grid-cols-4"
  >
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <img
          alt="modern"
          class="w-full rounded-xl shadow-lg"
          data-nimg="1"
          decoding="async"
          height="180"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=640&q=75 2x"
          style="color: transparent;"
          width="315"
        />
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-col justify-start items-start pt-4 pl-2"
      >
        <p
          class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance"
        >
          #1
        </p>
        <p
          class="text-tremor-title w-full font-bold text-xl capitalize mb-2 text-primary dark:text-dark-primary"
        >
          <a
            href="/all-projects"
          >
            modern
          </a>
        </p>
        <p
          class="text-tremor-default text-tremor-content flex-wrap font-light mb-6 dark:text-dark-tremor-content-romance"
        >
          As Uber works through a huge amount of internal management turmoil.
        </p>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <a
            href="/all-projects"
          >
            <button
              class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
            >
              <span
                class="tremor-Button-text text-tremor-default whitespace-nowrap"
              >
                <p
                  class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                >
                  View Project
                </p>
              </span>
            </button>
          </a>
          <div
            class="container-avatars min-w-[56px] flex justify-end flex-nowrap ml-3"
          >
            <div
              class="inline-flex rounded-full overflow-hidden relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
            >
              <img
                alt="Anne Marie"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="20"
                sizes="100%"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                style="color: transparent;"
                width="20"
              />
            </div>
            <div
              class="inline-flex rounded-full overflow-hidden relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
            >
              <img
                alt="Nick Daniel"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="20"
                sizes="100%"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                style="color: transparent;"
                width="20"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <img
          alt="scandinavian"
          class="w-full rounded-xl shadow-lg"
          data-nimg="1"
          decoding="async"
          height="180"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=640&q=75 2x"
          style="color: transparent;"
          width="315"
        />
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-col justify-start items-start pt-4 pl-2"
      >
        <p
          class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance"
        >
          #2
        </p>
        <p
          class="text-tremor-title w-full font-bold text-xl capitalize mb-2 text-primary dark:text-dark-primary"
        >
          <a
            href="/all-projects"
          >
            scandinavian
          </a>
        </p>
        <p
          class="text-tremor-default text-tremor-content flex-wrap font-light mb-6 dark:text-dark-tremor-content-romance"
        >
          As Uber works through a huge amount of internal management turmoil.
        </p>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <a
            href="/all-projects"
          >
            <button
              class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
            >
              <span
                class="tremor-Button-text text-tremor-default whitespace-nowrap"
              >
                <p
                  class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                >
                  View Project
                </p>
              </span>
            </button>
          </a>
          <div
            class="container-avatars min-w-[56px] flex justify-end flex-nowrap ml-3"
          >
            <div
              class="inline-flex rounded-full overflow-hidden relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
            >
              <img
                alt="Anne Marie"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="20"
                sizes="100%"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                style="color: transparent;"
                width="20"
              />
            </div>
            <div
              class="inline-flex rounded-full overflow-hidden relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
            >
              <img
                alt="Nick Daniel"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="20"
                sizes="100%"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                style="color: transparent;"
                width="20"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <img
          alt="minimalist"
          class="w-full rounded-xl shadow-lg"
          data-nimg="1"
          decoding="async"
          height="180"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=640&q=75 2x"
          style="color: transparent;"
          width="315"
        />
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-col justify-start items-start pt-4 pl-2"
      >
        <p
          class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance"
        >
          #2
        </p>
        <p
          class="text-tremor-title w-full font-bold text-xl capitalize mb-2 text-primary dark:text-dark-primary"
        >
          <a
            href="/all-projects"
          >
            minimalist
          </a>
        </p>
        <p
          class="text-tremor-default text-tremor-content flex-wrap font-light mb-6 dark:text-dark-tremor-content-romance"
        >
          Different people have different taste, and various types of music.
        </p>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <a
            href="/all-projects"
          >
            <button
              class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
            >
              <span
                class="tremor-Button-text text-tremor-default whitespace-nowrap"
              >
                <p
                  class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                >
                  View Project
                </p>
              </span>
            </button>
          </a>
          <div
            class="container-avatars min-w-[56px] flex justify-end flex-nowrap ml-3"
          >
            <div
              class="inline-flex rounded-full overflow-hidden relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
            >
              <img
                alt="Anne Marie"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="20"
                sizes="100%"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                style="color: transparent;"
                width="20"
              />
            </div>
            <div
              class="inline-flex rounded-full overflow-hidden relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
            >
              <img
                alt="Nick Daniel"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="20"
                sizes="100%"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                style="color: transparent;"
                width="20"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <img
          alt="gothic"
          class="w-full rounded-xl shadow-lg"
          data-nimg="1"
          decoding="async"
          height="180"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fprofile-project.webp&w=640&q=75 2x"
          style="color: transparent;"
          width="315"
        />
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-col justify-start items-start pt-4 pl-2"
      >
        <p
          class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance"
        >
          #4
        </p>
        <p
          class="text-tremor-title w-full font-bold text-xl capitalize mb-2 text-primary dark:text-dark-primary"
        >
          <a
            href="/all-projects"
          >
            gothic
          </a>
        </p>
        <p
          class="text-tremor-default text-tremor-content flex-wrap font-light mb-6 dark:text-dark-tremor-content-romance"
        >
          Why would anyone pick blue over pink? Pink is obviously a better color.
        </p>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <a
            href="/all-projects"
          >
            <button
              class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
            >
              <span
                class="tremor-Button-text text-tremor-default whitespace-nowrap"
              >
                <p
                  class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                >
                  View Project
                </p>
              </span>
            </button>
          </a>
          <div
            class="container-avatars min-w-[56px] flex justify-end flex-nowrap ml-3"
          >
            <div
              class="inline-flex rounded-full overflow-hidden relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
            >
              <img
                alt="Anne Marie"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="20"
                sizes="100%"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                style="color: transparent;"
                width="20"
              />
            </div>
            <div
              class="inline-flex rounded-full overflow-hidden relative border-2 border-white border-solid ml-[-10px] cursor-pointer hover:z-10"
            >
              <img
                alt="Nick Daniel"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="20"
                sizes="100%"
                src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                style="color: transparent;"
                width="20"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
